virttest/qemu_devices/qcontainer.py

```diff
@@ -3201,3 +3202,41 @@ def numa_hmat_cache_define_by_params(self, nodeid, params):
                     aobject=aobject, backend='hmat_type'))
 
         return hmat_cache_devs
+
+    def secret_object(self, secret_id, data=None, secret_format='raw',
+                      encrypt=True, inline=True, secret_file=None):
+        """
+        Generate secret object device with given data, will create a new file
+        when inline is False and file is None
+
+        :param secret_id: the id of the secret object
+        :param data: password, could be None when file is provided and
+                     inline is True
+        :param secret_format: raw or base64
+        :param encrypt: encrypt data when True and format is not raw
+        :param inline: append data=xxx if True and file=xxx if False
+        :param secret_file: could be None when password is provided or inline=False
+        :return: the secret object device QObject
+        """
+        params = {'id': secret_id}
+        if secret_format != 'raw':
+            params.update({'format': secret_format})
+            if not secret_file and encrypt:
+                data = base64.b64encode(data.encode()).decode()
+        if not inline:
+            if not secret_file:
+                suffix = '.secret'
+                secret_file = os.path.join(
+                    data_dir.get_tmp_dir(), '%s' % secret_id)
+                i = 1
+                file_dir_tmp = secret_file
+                while os.path.exists(file_dir_tmp + suffix):
+                    file_dir_tmp = secret_file + '_%s' % i
+                    i += 1
```

Comment:
If run test with a big loop(use inline=False), there will be a lot of secret_files, and no clean up action, it is a problem i think.
Discussed with @zhencliu  about this too, without adding clean up action,  we can use `secret_id` as the file name, and overwrite if the file exist.  How do you think about this ? @vivianQizhu

Reply:
@yanan-fu It is in `data_dir.get_tmp_dir` directory, will be removed somehow I suppose?

Reply:
It is the framework's tempoary dir with host, no clean up as far as i know.

Reply:
@yanan-fu I thought avocado will clean it, and normally after the job stop those temporary directory like named avocado_xxx would disappear, correct me if I was wrong.
```
# Destroy the data_dir.get_tmpdir ...
data_dir._tmp_tracker.unittest_refresh_dir_tracker()
```

Reply:
@vivianQizhu The `avocado_xxx` is same when launch `avocado run` one time - a test loop include multiple test cases, avocado will not clean it for every case (post process).

Reply:
@yanan-fu Okay, I get your point, it creates one more secret file for each test case(for spice) so you don't like too many files there when the job has many cases, I am okay anyway, will remove it. Thanks.